internal/test/e2e/tinkerbellHardwareQueue.go

```diff
@@ -0,0 +1,58 @@
+package e2e
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+)
+
+// Default timeout for Tink tests to poll for hardware.
+const hwPollingTimeout = 120 * time.Minute
+
+type hardwareQueue struct {
+	hws []*api.Hardware
+	mu  sync.Mutex
+}
+
+func (hwQu *hardwareQueue) reserveHardware(count int) ([]*api.Hardware, error) {
```

Comment:
What happens if, in a hypothetical world, hardware never becomes available? How does this gracefully terminate?